File: leetcode/two-sum/two-sum.cpp
//Runtime: 144 ms, faster than 33.43% of C++ online submissions for Two Sum.
//Memory Usage: 9.3 MB, less than 98.81% of C++ online submissions for Two Sum.
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result;
        for(int i = 0; i < nums.size(); i++){
            for (int j = i + 1; j < nums.size(); j++){
                if (target - nums[i] == nums[j]){
                    result.push_back(i);
                    result.push_back(j);
                    return result;
                } 
            }
        }
        return  result;
    }
};

File: leetcode/two-sum/two-sum.go
//Runtime: 36 ms, faster than 41.70% of Go online submissions for Two Sum.
//Memory Usage: 2.9 MB, less than 100.00% of Go online submissions for Two Sum.

func twoSum(nums []int, target int) []int {
	for i := 0; i < len(nums); i += 1{
		for j := i + 1; j < len(nums); j += 1{
			if nums[i] + nums[j] == target{
				return []int {i,j}
			}
		}
	}
	return []int {0,1}
}